Footer global component styling

ramram-mf opened the pull request:
# Description

This PR implements the global footer component styling.

Link to sample test page:
Related PRs/issues: #

sabrinang reviewed (changes requested):
**Desktop**
- Newsletter sign up: Can the button be located below the checkbox small text and not push the other elements over horizontally when selected?
- Align divider alignment/width at largest breakpoint (falls a bit short)
- Hover state for social icons (orange-100 `FF9E5F`)

**Mobile**
- Social icon spacing: keep them clustered like desktop as they look too spread apart with centre alignment

mmmavis commented on foundation_cms/templates/patterns/components/footer.html:
```diff
@@ -34,31 +34,38 @@
-            <a href="https://bsky.app/profile/mozilla.org" aria-label="Bluesky">
+            <a href="https://bsky.app/profile/mozilla.org" class="site-footer-ns__social-link" aria-label="Bluesky">
               <i class="icon-bluesky">
                 <svg width="24" height="23" viewBox="0 0 24 23" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M12 10.2734C10.913 8.15945 7.954 4.22045 5.202 2.27845C2.566 0.417447 1.561 0.739447 0.902 1.03845C0.139 1.38145 0 2.55345 0 3.24145C0 3.93145 0.378 8.89145 0.624 9.72045C1.439 12.4564 4.337 13.3804 7.007 13.0844C7.143 13.0644 7.282 13.0454 7.422 13.0284C7.284 13.0504 7.146 13.0 ...
```
Are we animating the SVG itself here? If not, could we link to the file instead of inlining it?

mmmavis replied:
Same with other SVG files used in this file. Let's link to the SVG files directly.

mmmavis commented on foundation_cms/templates/patterns/components/footer.html:
```diff
@@ -34,31 +34,38 @@
-            <a href="https://www.instagram.com/mozilla/" aria-label="Instagram">
+            <a href="https://www.instagram.com/mozilla/" class="site-footer-ns__social-link" aria-label="Instagram">
               <i class="icon-instagram">
                 <svg width="25" height="25" viewBox="0 0 25 25" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#clip0_677_6983)"><path d="M7.78008 0.557405C6.50328 0.617605 5.63138 0.821406 4.86908 1.12081C4.08028 1.42831 3.41158 1.84081 2.74628 2.50851C2.08108 3.17621 1.67128 3.84531 1.36608 4.63551C1.07068 5.39931 0.870482 6.27201 0.814082 7.54951C0.757682 8.82701 0.745182 9.2 ...
```
Are we animating the SVG itself here? If not, could we link to the file instead of inlining it?

ramram-mf commented:
> **Desktop**
> 
>     * Newsletter sign up: Can the button be located below the checkbox small text and not push the other elements over horizontally when selected?
> 
>     * Align divider alignment/width at largest breakpoint (falls a bit short)
> 
>     * Hover state for social icons (orange-100 `FF9E5F`)
> 
> 
> **Mobile**
> 
>     * Social icon spacing: keep them clustered like desktop as they look too spread apart with centre alignment

Hi @sabrinang I've addressed all of these but the newsletter signup submit button alignment in large viewports, I updated the styling to have a fixed width on the form inputs so that there are no layout shifts when displaying the hidden dropdowns. Do you think this would suffice in terms of a solution? Or should we explore the top-to-bottom layout in the fieldset and button wrappers. Since this component has this layout in the other application instance and that visual update would reflect site-wide.

mmmavis reviewed (approved):
🎉

sabrinang reviewed (approved):
Thanks for the update, looks good!
